src/openrct2/actions/GameActionResult.h

```diff
@@ -82,13 +82,13 @@ namespace GameActions
         // is still just uint32_t, this guarantees the data is associated with the correct type.
         template<typename T> void SetData(const T&& data)
         {
-            ResultData = std::forward<const T&&>(data);
+            ResultData = std::make_shared<T>(data);
         }
 
-        // This function will throw std::bad_any_cast if the type mismatches.
         template<typename T> T GetData() const
         {
-            return std::any_cast<T>(ResultData);
+            T* res = static_cast<T*>(ResultData.get());
+            return *res;
```

Comment:
We discussed this solution in the team chat. A problem with this is that it essentially throws out all type checks.

This solution is fine for Android as a stop-gap measure, but we would like it if the other platforms didn’t have to suffer from this. As such, can you put your changes behind an \#ifdef?